web/gds-user-ui/src/modules/dashboard/Certificate/registration.tsx

```diff
@@ -0,0 +1,153 @@
+import { SimpleDashboardLayout } from 'layouts';
+import { Box, Heading, HStack, VStack, useToast } from '@chakra-ui/react';
+import Card from 'components/ui/Card';
+import TestNetCertificateProgressBar from 'components/TestnetProgress/TestNetCertificateProgressBar.component';
+import FormButton from 'components/ui/FormButton';
+import useCertificateStepper from 'hooks/useCertificateStepper';
+import { FormProvider, useForm, useFormState } from 'react-hook-form';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { DevTool } from '@hookform/devtools';
+import { RootStateOrAny, useSelector } from 'react-redux';
+import _ from 'lodash';
+import { hasStepError } from '../../../utils/utils';
+
+import {
+  fieldNamesPerSteps,
+  certificateRegistrationValidationSchema,
+  getCertificateRegistrationDefaultValue
+} from './lib';
+
+const fieldNamesPerStepsEntries = () => Object.entries(fieldNamesPerSteps);
+
+const Certificate: React.FC = () => {
+  const { nextStep, previousStep } = useCertificateStepper();
+  const currentStep: number = useSelector((state: RootStateOrAny) => state.stepper.currentStep);
+  const lastStep: number = useSelector((state: RootStateOrAny) => state.stepper.lastStep);
+  const steps: number = useSelector((state: RootStateOrAny) => state.stepper.steps);
+  const hasReachSubmitStep: boolean = useSelector(
+    (state: RootStateOrAny) => state.stepper.hasReachSubmitStep
+  );
+  const toast = useToast();
+  const current = currentStep === lastStep ? lastStep - 1 : currentStep;
+
+  const methods = useForm({
+    defaultValues: getCertificateRegistrationDefaultValue(),
+    resolver: yupResolver(certificateRegistrationValidationSchema),
+    mode: 'onChange'
+  });
+
+  function getFieldValue(name: string) {
+    return _.get(methods.getValues(), name);
+  }
+
+  function isFormCompleted() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.every((n) => getFieldValue(n).toString());
+  }
+
+  function getCurrentFormValue() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.reduce((acc, n) => ({ ...acc, [n]: getFieldValue(n) }), {});
+  }
+
+  function hasErroredField() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.some((n: any) => methods.getFieldState(n).error);
+  }
+
+  function handleNextStepClick() {
+    if (currentStep === lastStep) {
+      if (hasStepError(steps)) {
+        toast({
+          position: 'top',
+          title: `Please fill all the required fields before submitting`,
```

Comment:
What do you think of this wording, @rebeccabilbro? The user receives this error message when moving from the review step to the submission step.

Reply:
Thanks for checking @masskoder ! I might modify it slightly, e.g.

```Please fill in all required fields before proceeding```

And hopefully the fields they missed will be indicated with asterisks or in red somehow?